### PR TITLE
tapping on iOS status bar should scroll inverted ScrollViews to their top

### PR DESCRIPTION
**Pull request by emilioicai**
## Summary

React Native ScrollViews are flipped upside down when the prop inverted is set to true. This is the root of a bug: tapping on the status bar in iOS should scroll the Flatlist up to the top but currently it does to the bottom.

The solution proposed is to detect natively if the ScrollView is inverted, on such case, prevent it from scrolling it to the beginning of the ScrollView (as a non-inverted ScrollView would do) and force a scroll to the end of it.

I've been careful enough not to force the scroll if the user explicitly selected not to do it or if it's happening in a nested ScrollView, as it is the default behaviour in iOS.

Fixes https://github.com/facebook/react-native/issues/21126

## Changelog

[iOS] [Fixed] - Inverted ScrollViews scroll to their bottom when the status bar is pressed

## Test Plan

- on iOS, add a ScrollView and put enough content to overflow the screen size so it can be scrolled
- add the prop `inverted={true}` to the ScrollView
- go to the screen the Scrollview is in and press the status bar
- it should scroll to top (previously it scrolled to the bottom)

v

### CODE REVIEW

File: React/Views/ScrollView/RCTScrollView.m [hunk 1]
```diff
@@ -829,6 +829,10 @@ - (BOOL)scrollViewShouldScrollToTop:(UIScrollView *)scrollView
       return NO;
     }
   }
+  if(self.inverted) {
```
Comment (hramos) [inline React/Views/ScrollView/RCTScrollView.m]:
```suggestion

  if (self.inverted) {
```

File: React/Views/ScrollView/RCTScrollView.m [hunk 2]
```diff
@@ -829,6 +829,10 @@ - (BOOL)scrollViewShouldScrollToTop:(UIScrollView *)scrollView
...
+    [self scrollToEnd:YES];
+    return NO;
+  }
```
Comment (hramos) [inline React/Views/ScrollView/RCTScrollView.m]:
```suggestion
  }

```

**Comment by Hirbod** [comment]
@emilioicai sorry for commenting an "old" ticket, but Expo SDK just released with RN 0.62.2 and that fix is included there so its the first time I can test it. Your change works fine so far, the direction works, but only when I scroll "up" at least 1px. If my chat scroll position is "0" a tap on the statusbar won't do anything for me now. If I just scroll up a bit, scroll to top (or here: to bottom since its inverted) works.

**Comment by Hirbod** [comment]
Here is a video (listen to audio please)
https://streamable.com/5pkkko

**Comment by PAJoeInc** [comment]
@Hirbod were you ever able to find a solution to this issue? I am also experiencing the same thing in RN 0.63.2

**Comment by Hirbod** [comment]
@PAJoeInc no - unfortunately not, since nobody responded to my message here.

**Comment by Hirbod** [comment]
I think I will have to write a new ticket

**Comment by jake-stewart** [comment]
very annoying. i want the old behaviour as how discord handles it.

**Pull request closed without merging**